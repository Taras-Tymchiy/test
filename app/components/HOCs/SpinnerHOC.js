// @flow
import * as React from 'react';
import { View, StyleSheet, ActivityIndicator } from 'react-native';

type SpinnerProps = {
  spinner?: boolean,
  error?: Error,
};

export default function SpinnerHOC<Props: SpinnerProps, Component: React.ComponentType<$Diff<Props, SpinnerProps>>>(Comp: Component): React.ComponentType<Props>  {
  return ({ spinner, ...props }: Props) => (
    <View style={{ flex: 1 }}>
      <Comp {...props} />
      {spinner &&
        <View
          style={[
            StyleSheet.absoluteFill,
            { backgroundColor: 'rgba(0, 0, 0, 0.6)', justifyContent: 'center' }
          ]}
        >
          <ActivityIndicator size="large" />
        </View>}
    </View>
  );
};