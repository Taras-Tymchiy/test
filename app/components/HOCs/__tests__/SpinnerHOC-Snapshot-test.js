import React from 'react';
import renderer from 'react-test-renderer';
import SpinnerHOC from '../SpinnerHOC';
import { View } from 'react-native';

it('renders a SpinnerHOC wrapped View with spinner using Snapshots', () => {
  const WrappedView = SpinnerHOC(View);
  expect(renderer.create(
    <WrappedView spinner={true} />
  )).toMatchSnapshot();
});
  
it('renders a SpinnerHOC wrapped View without spinner using Snapshots', () => {
  const WrappedView = SpinnerHOC(View);
  expect(renderer.create(
    <WrappedView />
  )).toMatchSnapshot();
});
