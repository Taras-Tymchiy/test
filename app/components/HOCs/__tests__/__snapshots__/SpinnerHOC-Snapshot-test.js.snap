// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a SpinnerHOC wrapped View with spinner using Snapshots 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "justifyContent": "center",
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;

exports[`renders a SpinnerHOC wrapped View without spinner using Snapshots 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
</View>
`;
